arcee/api.py

```diff
@@ -82,26 +125,85 @@ def start_pretraining(pretraining_name: str, corpus: str, base_model: str) -> No
 
     data = {"pretraining_name": pretraining_name, "corpus_name": corpus, "base_model": base_model}
 
-    return make_request("post", Route.pretraining+"/startTraining", data)
+    return make_request("post", Route.pretraining + "/startTraining", data)
+
+def mergekit_yaml(
+    merging_name: str,
+    merging_yaml_path: str
+) -> None:
+    """
+    Start merging models
+
+    Args:
+        merging_name (str): The name of the merging job
+        merging_yaml (str): The yaml file containing the merging instructions
+    """
 
-def start_merging(merging_name: str, arcee_models: Optional[List[str]] = None, hf_models: Optional[List[str]] = None, qa_set_ids: Optional[List[str]] = None, general_evals: Optional[List[str]] = None, base_model: Optional[str] = None, merge_method: Optional[str] = None, time_budget: int = 1) -> None:
+    if not merging_yaml_path.endswith(".yaml"):
+        raise Exception("The merging yaml file must be a .yaml file")
+
+    if not os.path.exists(merging_yaml_path):
+        raise Exception(f"The merging yaml file {merging_yaml_path} does not exist")
+
+    with open(merging_yaml_path, 'r') as file:
+        merging_yaml = yaml.safe_load(file)
+    
+        data = {
+            "merging_name": merging_name,
+            "best_merge_yaml": str(merging_yaml)
+        }
+
+        return make_request("post", Route.merging + "/startMerging", data)
+
+def mergekit_evolve(
+    merging_name: str,
+    wandb_key: str = None,
+    arcee_aligned_models: Optional[List[str]] = None,
+    arcee_merged_models: Optional[List[str]] = None,
+    arcee_pretrained_models: Optional[List[str]] = None,
+    hf_models: Optional[List[str]] = None,
+    arcee_eval_qa_set_names_and_weights: Optional[List[dict]] = None,
+    general_evals_and_weights: Optional[List[dict]] = None,
+    base_model: Optional[str] = None,
+    merge_method: Optional[str] = None,
+    target_compute: str = None,
+    capacity_id: str = None,
+    time_budget_secs: int = 1,
+) -> None:
     """
     Start merging models
 
     Args:
         merging_name (str): The name of the merging job
-        arcee_models (list): A list of ARCEE models to merge
+        arcee_aligned_models (list): A list of ARCEE models to merge
+        arcee_merged_models (list): A list of ARCEE models already merged
+        arcee_pretrained_models (list): A list of pretrained ARCEE models
         hf_models (list): A list of Hugging Face models to merge
-        qa_set_ids (list): A list of QA sets to merge
-        general_evals (list): A list of general evaluations to merge
+        eval_qa_set_names_and_weights (list): A list of QA set names to merge
+        general_evals_and_weights (list): A list of general evaluations to merge
         base_model (str): The name of the base model to use
         merge_method (str): The merging method to use
```

Comment:
Can we point to the list of available merging methods?  Also a sensible default might make it easier to use.